polars/polars-io/src/lib.rs

```diff
@@ -170,7 +172,6 @@ pub(crate) fn finish_reader<R: ArrowReader>(
 
 /// Check if the path is a cloud url.
 pub fn is_cloud_url<P: AsRef<Path>>(p: P) -> bool {
-    p.as_ref().starts_with("s3://")
-        || p.as_ref().starts_with("file://")
-        || p.as_ref().starts_with("gcs://")
+    let path = p.as_ref();
+    return CloudType::from_str(&path.to_string_lossy()).is_ok();
```

Comment:
CloudType also supports `File` so I think we should pattern match here.

Reply:
That File inside the enum refers to the URL using "file://" scheme. In the case of the "file://" scheme the could should still be sent to the Cloud code.